helm: fix large number handling

justin0u0 opened the pull request:
Handle large values that may be turned into scientific notation, which
cause parsing error in Cilium.

Fixes: #34010

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

gandro commented on install/kubernetes/cilium/templates/cilium-configmap.yaml:
```diff
@@ -431,12 +431,12 @@ data:
   # bpf-nat-global-max specified the maximum number of entries in the
   # BPF NAT table.
-  bpf-nat-global-max: "{{ .Values.bpf.natMax | int }}"
+  bpf-nat-global-max: "{{ .Values.bpf.natMax | int | quote }}"
```
On closer inspection, the use of `quote` here is wrong. Either use `foo: "{{ .Values.bar | int}}"`  or use `foo: {{ .Values.bar | int | quote}}` - but not both.

justin0u0 replied:
Ah, I didn't notice that, thanks!

I fixed the above lines to make it consistent with other part of the code.

Fixed in https://github.com/cilium/cilium/pull/37670/commits/4708b740f6f60f2671187d1b304f3dd5ba07787d

gandro commented on install/kubernetes/cilium/templates/cilium-configmap.yaml:
```diff
@@ -431,12 +431,12 @@ data:
   # bpf-neigh-global-max specified the maximum number of entries in the
   # BPF neighbor table.
-  bpf-neigh-global-max: "{{ .Values.bpf.neighMax | int }}"
+  bpf-neigh-global-max: "{{ .Values.bpf.neighMax | int | quote }}"
```
Same as above

justin0u0 replied:
Fixed in https://github.com/cilium/cilium/commit/4708b740f6f60f2671187d1b304f3dd5ba07787d, thanks!

justin0u0 committed:
helm: fix large number handling

Handle large values that may be turned into scientific notation, which
cause parsing error in Cilium.

Signed-off-by: justin0u0 <mail@justin0u0.com>
